python_modules/libraries/dagster-databricks/dagster_databricks/databricks_step_main.py

```diff
@@ -0,0 +1,131 @@
+'''
+
+import os
+import pickle
+import site
+import sys
+import tempfile
+import zipfile
+
+from dagster.core.execution.plan.external_step import PICKLED_EVENTS_FILE_NAME, run_step_from_ref
+from dagster.serdes import serialize_value
+
+
+# This won't be set in Databricks but is needed to be non-None for the
+# Dagster step to run.
+if 'DATABRICKS_TOKEN' not in os.environ:
+    os.environ['DATABRICKS_TOKEN'] = ''
+
+
+def setup_s3_storage(scope, credentials):
+    '''Obtain AWS credentials from Databricks secrets and export so both Spark and boto can use them.
+    '''
+    # dbutils is globally defined in the Databricks runtime
+    access_key = dbutils.secrets.get(  # noqa  # pylint: disable=undefined-variable
+        scope=scope, key=credentials['access_key_key']
+    )
+    secret_key = dbutils.secrets.get(  # noqa  # pylint: disable=undefined-variable
+        scope=scope, key=credentials['secret_key_key']
+    )
+
+    # Spark APIs will use this.
+    # See https://docs.databricks.com/data/data-sources/aws/amazon-s3.html#alternative-1-set-aws-keys-in-the-spark-context.
+    # sc is globally defined in the Databricks runtime and points to the Spark context
+    sc._jsc.hadoopConfiguration().set(  # noqa  # pylint: disable=undefined-variable,protected-access
+        'fs.s3n.awsAccessKeyId', access_key
+    )
+    sc._jsc.hadoopConfiguration().set(  # noqa  # pylint: disable=undefined-variable,protected-access
+        'fs.s3n.awsSecretAccessKey', secret_key
+    )
+
+    # Boto will use these.
+    os.environ['AWS_ACCESS_KEY_ID'] = access_key
+    os.environ['AWS_SECRET_ACCESS_KEY'] = secret_key
+
+
+def setup_adls2_storage(scope, credentials):
+    '''Obtain an Azure Storage Account key from Databricks secrets and export so Spark can use it.
+    '''
+    # dbutils is globally defined in the Databricks runtime
+    storage_account_key = dbutils.secrets.get(  # noqa  # pylint: disable=undefined-variable
+        scope=scope, key=credentials['storage_account_key_key']
+    )
+    # Spark APIs will use this.
+    # See https://docs.microsoft.com/en-gb/azure/databricks/data/data-sources/azure/azure-datalake-gen2#--access-directly-using-the-storage-account-access-key
+    # sc is globally defined in the Databricks runtime and points to the Spark context
+    sc._jsc.hadoopConfiguration().set(  # noqa  # pylint: disable=undefined-variable,protected-access
+        'fs.azure.account.key.{}.dfs.core.windows.net'.format(credentials['storage_account_name']),
+        storage_account_key,
+    )
+
+
+def setup_storage(step_run_ref):
+    '''Setup any storage required by the run.
+
+    At least one of S3 or ADLS2 storage should be provided in config, so that the run can
+    save intermediate files to a location accessible by the original process which launched
+    the job.
+    '''
+    storage = step_run_ref.environment_dict['resources']['pyspark_step_launcher']['config'][
+        'storage'
+    ]
```

Comment:
Can we do a `check.invariant` here to make sure that the storage here matches what's under `step_run_ref.environment_dict['storage']`?